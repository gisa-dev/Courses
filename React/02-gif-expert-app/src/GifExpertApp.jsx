import {useState} from 'react';
import {AddCategory, GifGrid} from './components';

const GifExpertApp = () => {
	const [categories, setCategories] = useState(['One Piece', 'Fairy Tail']);

	const onAddCategory = (newCategory) => {
		if (categories.includes(newCategory)) return;

		setCategories([newCategory, ...categories]);
		// setCategories(cat => [...cat, 'Dragon Ball']);
	};

	return (
		<>
			{/* Title */}
			<h1>GifExpertApp</h1>

			{/* Input */}

			<AddCategory onNewCategory={(value) => onAddCategory(value)} />

			{categories.map((category) => (
				<GifGrid key={category} category={category} />
			))}
		</>
	);
};

export default GifExpertApp;
