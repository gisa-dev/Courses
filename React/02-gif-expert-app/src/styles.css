* {
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  background-color: rgb(234, 234, 234);
}

body {
  padding: 60px;
}

input {
  background-color: white;
  border-radius: 0.3rem;
  border: 0.06rem solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 0.6rem 0.9rem;
  width: 100%;
  margin-bottom: 0.6rem;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin-bottom: 0.3rem;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  gap: 1.25rem;
}

.card {
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 0.3rem;
  box-shadow: 0rem 0.1rem 0.3rem rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
   
  justify-content: center;
  gap: 0.6rem;
  overflow: hidden;
}

.card p {
  background-color: white;
  flex: 1;
  font-size: 1.5rem;
  margin-top: 1rem;
  padding: 1rem 4rem 0rem 4rem;
  text-align: center;
}

.card img {
  width: 100%;
  height: 100%;
}

