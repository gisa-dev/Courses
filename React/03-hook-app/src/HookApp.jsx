
const HookApp = () => {
  return (
    <div>HookApp</div>
  )
}

export default HookApp