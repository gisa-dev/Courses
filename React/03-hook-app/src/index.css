body {
	padding: 20px;
	background-color: #252525;
	color: #fcfcfc;
}

button {
	margin-right: 10px;
}
