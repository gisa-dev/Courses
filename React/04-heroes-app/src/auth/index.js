export { AuthContext, AuthProvider, authReducer } from './context';
export { LoginPage } from './pages';
