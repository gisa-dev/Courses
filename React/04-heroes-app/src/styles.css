@tailwind base;
@tailwind components;
@tailwind utilities;

.title {
	@apply py-5 mb-5 text-4xl font-semibold border-b-4 border-slate-300;
}
