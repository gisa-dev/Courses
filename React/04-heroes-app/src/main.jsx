import React from 'react';
import { createRoot } from 'react-dom/client';
import { BrowserRouter } from 'react-router-dom';
import './styles.css';
import { HeroesApp } from './HeroesApp';

createRoot(document.getElementById('root')).render(
	<React.StrictMode>
		<BrowserRouter>
			<HeroesApp />
		</BrowserRouter>
	</React.StrictMode>,
);
