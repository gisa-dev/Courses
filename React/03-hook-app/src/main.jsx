import React from 'react';
import { createRoot } from 'react-dom/client';
import './index.css';
import HookApp from './HookApp';
import { BrowserRouter } from 'react-router-dom';

createRoot(document.getElementById('root')).render(
	// <React.StrictMode>
	<BrowserRouter>
		<HookApp />
	</BrowserRouter>,
	// </React.StrictMode>,
);
