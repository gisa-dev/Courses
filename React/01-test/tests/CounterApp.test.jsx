import {render, screen} from '@testing-library/react';
import {CounterApp} from '../src/CounterApp';

describe('Test in <CounterApp/>', () => {
	const initialValue = 20;

	test('should match the snapshot', () => {
		const {container} = render(<CounterApp value={initialValue} />);
		expect(container).toMatchSnapshot();
	});

	test('should show the initial value of 100', () => {
		render(<CounterApp value={100} />);
		expect(screen.getByText(100)).toBeTruthy();
	});
});
