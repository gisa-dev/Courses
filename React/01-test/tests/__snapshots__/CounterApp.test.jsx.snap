// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <CounterApp/> should match the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    20
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
